### PR TITLE
Add usage information to README

### PR DESCRIPTION
**Pull request by vicsn**
I set the blog post URL to: https://blog.summitto.com/posts/deterministic_pgp_deep_post/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,20 +2,39 @@
...
+  * [Assumptions](#assumptions)
+  * [Upgrading hardware tokens](#upgrading-hardware-tokens)
+  * [Swapping secret data to disk](#swapping-secret-data-to-disk)
+* [Static analysis](#static-analysis)
+  * [CppCheck](#cppcheck)
+  * [Clang Tidy](#clang-tidy)
+
+This repository provides the source for a utility which can be used for
+creating PGP keys using [libsodium](https://download.libsodium.org/doc/
+"Introduction - Libsodium documentation") (for ed- and curve25519 keys) or
```
Comment (rockhouse) [inline README.md:22]: I would say "ed25519 and curve25519"

File: README.md [hunk 2]
```diff
@@ -2,20 +2,39 @@
...
+[Crypto++](https://www.cryptopp.com/ "Crypto++ Library | Free C++ Class Library
+of Cryptographic Schemes") (for nist-p256 and RSA keys).
```
Comment (rockhouse) [inline README.md:24]: Nist

File: README.md [hunk 3]
```diff
@@ -2,20 +2,39 @@
...
+
+Generating keys will also output a 64-character hexadecimal seed value, which,
+in combination with a chosen passphrase, can be used to recreate the exact same
```
Comment (rockhouse) [inline README.md:27]: and creation date

File: README.md [hunk 4]
```diff
@@ -2,20 +2,39 @@
 
 # PGP key generation
```
Comment (rockhouse) [inline README.md:3]: Deterministic PGP key generation

File: README.md [hunk 5]
```diff
@@ -51,72 +70,192 @@ Security](https://radicallyopensecurity.com/) in November 2019:
...
+ramdisk will be set up in the `~/ramdisk folder`. This allows you to indicate a
+`keyfile` in the ramdisk (e.g. `~/ramdisk/keyfile.asc`) in order to ensure no
+sensitive key material is stored permanently. 
 
-## Generating new keys
+### Prepare pen and paper or Cryptosteel
 
-Please read all of the steps below thoroughly before actually starting to
-generate a key.
+The main purpose of the pgp key generation utility is that it allows you to
```
Comment (rockhouse) [inline README.md:94]: pgp-key-generation

File: README.md [hunk 6]
```diff
@@ -51,72 +70,192 @@ Security](https://radicallyopensecurity.com/) in November 2019:
...
+recover your PGP keys with very little data. If you lose access to your keys in
+the future and want to recover them, you will need:  
+- (1) the key creation time  
+- (2) the recovery seed which is exported by the utility
```
Comment (rockhouse) [inline README.md:98]: or mnemonic

File: README.md [hunk 7]
```diff
@@ -51,72 +70,192 @@ Security](https://radicallyopensecurity.com/) in November 2019:
...
+If you didn't specify additional flags, the program will ask you to fill in a
+number of details for your key:  
+- type of your key  
+- firstname and lastname  
+- email address  
+- signature creation time, for example: `2019-12-31 23:59:59`  
+  As mentioned above, please store this date in order to be able to recover
+  your key.
+- signature expiry time, for example: `2020-03-31 23:59:59`  
+- key creation time, for example: `2019-12-31 23:59:59`
```
Comment (rockhouse) [inline README.md:140]: for convenience can be the same as "signature creation time"

File: README.md [hunk 8]
```diff
@@ -51,72 +70,192 @@ Security](https://radicallyopensecurity.com/) in November 2019:
...
+  password. The resulting encrypted recovery seed will be almost twice as long,
+  but will also help to assure confidentiality and integrity. (For more
+  information about the encryption used, see the
+  [Libsodium](https://libsodium.gitbook.io/doc/public-key_cryptography/authenticated_encryption#purpose)
+  documentation.
+- Next, the program will convert the recovery seed into a mnemonic recovery
+  phrase in the language of your choice! 
+- Make sure you store the mnemonic (which is your recovery seed) in a secure
+  place. **This is your only chance to backup the mnemonic.**
+- You can now export the subkey pairs to your security token:
```
Comment (rockhouse) [inline README.md:163]: Maybe mention before NitroKey, Solokeys and Yubikey and that this is actually preferred way of "using" the PGP key?

**Review by rockhouse — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (vicsn) [reply README.md]: Fixed in 91ce89e2b2fd04770f8cec3cb24f9a8548c1b463

File: README.md (hunk 2 above)
Comment (vicsn) [reply README.md]: Fixed in 91ce89e2b2fd04770f8cec3cb24f9a8548c1b463

File: README.md (hunk 3 above)
Comment (vicsn) [reply README.md]: Fixed in 91ce89e2b2fd04770f8cec3cb24f9a8548c1b463

File: README.md (hunk 4 above)
Comment (vicsn) [reply README.md]: Fixed in 91ce89e2b2fd04770f8cec3cb24f9a8548c1b463

File: README.md (hunk 5 above)
Comment (vicsn) [reply README.md]: Fixed in 91ce89e2b2fd04770f8cec3cb24f9a8548c1b463

File: README.md (hunk 6 above)
Comment (vicsn) [reply README.md]: Fixed in 91ce89e2b2fd04770f8cec3cb24f9a8548c1b463 , using `mnemonic recovery phrase` now

File: README.md (hunk 7 above)
Comment (vicsn) [reply README.md]: Fixed in 91ce89e2b2fd04770f8cec3cb24f9a8548c1b463

**Review by rockhouse — APPROVED** [review]

**Pull request merged**